base/sparse/sparsevector.jl

```diff
@@ -25,15 +25,33 @@ end
 SparseVector(n::Integer, nzind::Vector{Ti}, nzval::Vector{Tv}) where {Tv,Ti} =
     SparseVector{Tv,Ti}(n, nzind, nzval)
 
+const AbstractSparseVectorUnion{T<:Number} =
+    Union{AbstractSparseVector{T},
+          SubArray{T,1,<:AbstractSparseArray,Tuple{Base.Slice{Base.OneTo{Int64}},Int64},false}}
+
 ### Basic properties
 
 length(x::SparseVector) = x.n
 size(x::SparseVector) = (x.n,)
 nnz(x::SparseVector) = length(x.nzval)
 countnz(x::SparseVector) = countnz(x.nzval)
 count(x::SparseVector) = count(x.nzval)
+
 nonzeros(x::SparseVector) = x.nzval
+function nonzeros(x::SubArray{T,1,<:AbstractSparseArray,Tuple{Base.Slice{Base.OneTo{Int64}},Int64},false} where T)
+    rowidx, colidx = parentindexes(x)
+    A = parent(x)
+    @inbounds y = view(A.nzval, A.colptr[colidx]:A.colptr[colidx + 1] - 1)
```

Comment:
`parent` isn't always guaranteed to have a `.nzval` or `.colptr` field if it's only required to be an `AbstractSparseArray`